code/drasil-lang/Language/Drasil/Classes.hs

```diff
@@ -154,16 +156,18 @@ class UnitEq u where
   uniteq :: Lens' u UDefn
 
 -----------------------------------------------------
--- TODO: It is ok to be able to view a (defining?) 'Relation', but not necessarily
--- to 'set' it,  as it might just not be settable. So Getter it is.
+
 -- | Has a function that turns into a 'Relation'.
 class ExprRelat c where
   -- | Holds a relation.
   relat :: c -> Relation
+  -- TODO: rename to `toRelat`/`asRelat`? essentially saying "this is an idea that we can convert into a logical equivalence checking relation"
```

Comment:
My comment is outdated here, but it's still true that I'm changing the meaning of ExprRelat.

It's no longer much of a "Relation", it's more so just a "is this a thing that we can encode as an expression/relation?".

For example, things that just contain UIDs to variables will be just `sy UID`, while things containing defining expressions will be `sy UID $= definingExpr`.

This ExprRelat should likely be renamed, but I couldn't think of a better name.

Reply:
we need to look at all uses to see what is going on. From your description, it's almost like it's being *abused* to get two different features!

Reply:
Sounds good.

We use it as follows right now:
* For TM definitions, since we have manually set equations and relations (not using ModelKinds enough here really), we use `relat <QD>` to get the equations related to EquationalModels. I think this one can be nicely minimized/removed by having a tighter integration with the ModelKinds.
* MultiDefn: for the `toDispExpr`, we intentionally don't break stable by using `relat` to display the whole relation
* In the last equation of a few "equation derivation lists".
* 
```
drasil-docLang/Drasil/DocumentLanguage/Definitions.hs:  buildDescription v u (relat g) m []) : fs
drasil-docLang/Drasil/DocumentLanguage/Definitions.hs:  foldr (\x -> buildDescription v u x m) [] [relat i]) : fs
```
This one was actually in a TODO note for me to look into today.

Reply:
I think we should strive to remove `relat` and `ExprRelat` entirely. We would need to create functions that implement similar functionality, on top of the existing classes.

Reply:
I think we can do that. The TMs, I think we can minimize much of it with a tighter integration with the ModelKinds. It seems that much of the current usage in the "last equation of a few 'equation derivation lists'" should really be converted into DisplayExprs. For the MultiDefn, we can just move the logic of relat into the DisplayExpr. I'll just need to inspect the Definitions.hs usage I mentioned but I think it will be doable.

Reply:
Sounds good.